File: src/main/java/no/nav/veilarboppfolging/repository/entity/OppfolgingsperiodeEntity.java
package no.nav.veilarboppfolging.repository.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Value;
import no.nav.veilarboppfolging.oppfolgingsbruker.StartetAvType;
import no.nav.veilarboppfolging.oppfolgingsbruker.inngang.OppfolgingStartBegrunnelse;

import javax.annotation.Nullable;
import java.time.ZonedDateTime;
import java.util.List;
import java.util.UUID;

@Value
@Builder(toBuilder = true)
@AllArgsConstructor
public class OppfolgingsperiodeEntity {
    UUID uuid;
    String aktorId;
    /**
     * Enten veileder-ident eller stringen "System" hvis avsluttet automatisk
     */
    String avsluttetAv;
    ZonedDateTime startDato;
    ZonedDateTime sluttDato;
    String begrunnelse;
    List<KvpPeriodeEntity> kvpPerioder;
    OppfolgingStartBegrunnelse startetBegrunnelse;
    @Nullable
    String startetAv;
    @Nullable
    StartetAvType startetAvType;
}

File: src/main/kotlin/no/nav/veilarboppfolging/repository/entity/OppfolgingsperiodeEntity.kt
package no.nav.veilarboppfolging.repository.entity

import no.nav.veilarboppfolging.oppfolgingsbruker.StartetAvType
import no.nav.veilarboppfolging.oppfolgingsbruker.inngang.OppfolgingStartBegrunnelse
import no.nav.veilarboppfolging.oppfolgingsbruker.utgang.AvregistreringsType
import java.time.ZonedDateTime
import java.util.UUID

data class OppfolgingsperiodeEntity(
    val uuid: UUID,
    val aktorId: String,
    /**
     * Enten veileder-ident eller stringen "System" hvis avsluttet automatisk
     */
    val avsluttetAv: String? = null,
    val startDato: ZonedDateTime,
    val sluttDato: ZonedDateTime? = null,
    val begrunnelse: String? = null,
    val kvpPerioder: List<KvpPeriodeEntity>,
    val startetBegrunnelse: OppfolgingStartBegrunnelse? = null,
    val startetAv: String? = null,
    val startetAvType: StartetAvType? = null,
    val avregistreringsType: AvregistreringsType? = null,
)